backend/lib/calculate/adequacy.py

```diff
@@ -69,92 +67,80 @@ def _fetch_addresses_from_ids(address_ids, engine):
 
 
 @timed
-def _find_addresses_near_service_area(
-    service_area_polygon,
-    providers
-):
-    """
-    Find providers within the given radius of the service area polygon.
-
-    This method reduces the number of distance calulations required by `calculate_adequacies`.
-    """
-    # TODO - Investigate ways to improve performance knowing that the polygons are rectangles
-    # (After switch to geometry datatype.)
-    # TODO - Investigate use of minmax latlon instead of complete polygons.
-    path = descartes.PolygonPatch(service_area_polygon).get_path()
-    return list(itertools.compress(
-        providers,
-        path.contains_points(
-            [(provider['longitude'], provider['latitude']) for provider in providers]
-        )
-    ))
-
-
-@timed
-def _fetch_service_area_polygons_db(
+def _get_addresses_to_check_by_service_area(
     service_area_ids,
+    addresses,
     radius_in_meters,
     engine=connect.create_db_engine()
 ):
     """
-    Find service areas and all providers within the given radius of the service area polygon.
+    Find addresses near each service area.
 
     This method reduces the number of distance calulations required by `calculate_adequacies`.
+
+    Returns a mapping service_area_id --> list of relevant addresses.
     """
-    # TODO - Add ST_ENVELOPE to force square bounding box after switch to geomery.
+    addresses_to_check_by_service_area = collections.defaultdict(list)
+
     service_area_id_list = '(VALUES' + ', '.join([
         "('{}')".format(_id) for _id in service_area_ids
     ]) + ')'
 
+    address_values = [
+        '({idx}, ST_SetSRID(ST_Point({longitude}, {latitude}), 4326)::geography)'.format(
+            idx=idx,
+            longitude=address['longitude'],
+            latitude=address['latitude']
+        )
+        for idx, address in enumerate(addresses)
+    ]
+
+    temp_table_name = ''.join(random.choice(string.ascii_lowercase) for _ in range(10))
```

Comment:
can we not simply use uuid or something to create a random string? :)

Reply:
That's what I tried at first -- UUIDs have characters that are invalid for Postgres table names. We could remove these but I figure that means we're losing the benefit of UUIDs.

Reply:
Ok, maybe abstract the function as _randomize_name() or something?

Reply:
Done.